provider/secret_descriptor.go

```diff
@@ -14,6 +15,12 @@ import (
 // An RE pattern to check for bad paths
 var badPathRE = regexp.MustCompile("(/\\.\\./)|(^\\.\\./)|(/\\.\\.$)")
 
+// An RE pattern to check for valid file permission
+var validFilePermissionRE = regexp.MustCompile("^[0-7]{4}$")
```

Comment:
Should we perform validation to ensure that the passed perms actually allow the driver to write the secret or do we just let it fail? https://github.com/aws/secrets-store-csi-driver-provider-aws/blob/5f41c0004924d6e322eccd38432eb8a936a9b44a/server/server.go#L318-L326

Say they set `0444` as the file permission, the write will fail, should we pre-empt this failure in the validation?